vignettes/opencl.Rmd

```diff
@@ -0,0 +1,117 @@
+---
+title: "Using CmdStanR to run Stan on the GPU with OpenCL"
+author: "Rok Češnovar and Jonah Gabry"
+output: 
+  rmarkdown::html_vignette:
+    toc: true
+    toc_depth: 4
+params:
+  EVAL: !r identical(Sys.getenv("CMDSTANR_OPENCL_TESTS"), "true")
+vignette: >
+  %\VignetteIndexEntry{Using CmdStanR to run Stan on the GPU with OpenCL}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r child="children/settings-knitr.Rmd"}
+```
+
+## Introduction
```

Comment:
I need to add a bit more context here and there but the general flow is what I imagined. Any comments welcome.

Reply:
@rok-cesnovar if I may chip in. What I expect to see in this sort of introduction is **how** and **when** OpenCL is useful for my Stan models? When N is higher than? or when the predictors are this much? Or when I have a hierarchical model with this and that? Or when I have a GP with big data?

And also when OpenCL is **not** recommended (possibly due to the overhead not being negligible than CPU backend)?

Reply:
Thanks for the input. Feel free to chime in at any time.

Reply:
For example [this model](https://github.com/storopoli/R_Scripts/blob/master/Stan/hierarchical_models/ncp_standard.stan) in my Dell G5 (i-7 12-core 16GB RAM CUDA 11.2 RTX2060 6GB) takes 2.2s and using OpenCL takes 586s (4 chains running in parallel in both). The full code for the testes can be found [here](https://github.com/storopoli/R_Scripts/tree/master/Stan/GPU)